### PR TITLE
Add ctl_tier_cfg struct

### PR DESCRIPTION
**Pull request by PatKamin**
Represent arguments in functions related to parsing environment variable in a more concise way.

Requires:
- [x] #590
### Types of changes

- [ ] Bugfix (non-breaking change which fixes issue linked in Description above)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (correction or otherwise)
- [ ] Cosmetics (whitespace, appearance)
- [x] Other

### Checklist

- [x] Code compiles without errors
- [x] All tests pass locally with my changes (see TESTING section in CONTRIBUTING file)
- [ ] Created tests which will fail without the change (if possible)
- [ ] Extended the README/documentation (if necessary)
- [ ] All newly added files have proprietary license (if necessary)
- [ ] All newly added files are referenced in MANIFEST files (if necessary)

## Further comments

---
This change is [Reviewable](https://reviewable.io/reviews/memkind/memkind/589)

### CODE REVIEW

**Review by bratpiorka — COMMENTED** [review]
> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/589)* status: all files reviewed, 2 unresolved discussions (waiting on @PatKamin)

___
*[tiering/ctl.h, line 13 at r1](https://reviewable.io/reviews/memkind/memkind/589#-MWTfTDg0H_9dPuNmnn4:-MWTfTDg0H_9dPuNmnn5:b-pyy2w8) ([raw file](https://github.com/memkind/memkind/blob/813a19d1d926e0c48d3eaa76d0ae31a616d829c6/tiering/ctl.h#L13)):*
> ```objc
> #endif
> 
> typedef struct tier_skeleton {
> ```

frightening name :) maybe _description? or _desc?
___
*[tiering/memtier.c, line 147 at r1](https://reviewable.io/reviews/memkind/memkind/589#-MWTflTm5AN7TcVbEgJr:-MWTflTm5AN7TcVbEgJs:b-gswdxv) ([raw file](https://github.com/memkind/memkind/blob/813a19d1d926e0c48d3eaa76d0ae31a616d829c6/tiering/memtier.c#L147)):*
> ```C
>     }
> 
>     memkind_t kind = get_kind(tier.kind_name, tier.pmem_path, tier.pmem_size);
> ```

pass just whole structure here. Note that you should probably move this function to ctl in future. Please talk with Michal about possible refactoring of this function - he has some vision of this

**Review by PatKamin — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/589)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @bratpiorka and @PatKamin)

___
*[tiering/memtier.c, line 147 at r1](https://reviewable.io/reviews/memkind/memkind/589#-MWTflTm5AN7TcVbEgJr:-MWTxtcOEbyitub6iATu:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/813a19d1d926e0c48d3eaa76d0ae31a616d829c6/tiering/memtier.c#L147)):*
<details><summary><i>Previously, bratpiorka (Rafał Rudnicki) wrote…</i></summary><blockquote>

pass just whole structure here. Note that you should probably move this function to ctl in future. Please talk with Michal about possible refactoring of this function - he has some vision of this
</blockquote></details>

Done.

**Review by PatKamin — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/589)* status: 0 of 3 files reviewed, 2 unresolved discussions (waiting on @bratpiorka)

___
*[tiering/ctl.h, line 13 at r1](https://reviewable.io/reviews/memkind/memkind/589#-MWTfTDg0H_9dPuNmnn4:-MWU3K8JBIp34WWRm0gA:b-z8l4u2) ([raw file](https://github.com/memkind/memkind/blob/813a19d1d926e0c48d3eaa76d0ae31a616d829c6/tiering/ctl.h#L13)):*
<details><summary><i>Previously, bratpiorka (Rafał Rudnicki) wrote…</i></summary><blockquote>

frightening name :) maybe _description? or _desc?
</blockquote></details>

_draft ?

File: tiering/ctl.h [hunk 1]
```diff
@@ -2,10 +2,21 @@
 /* Copyright (C) 2021 Intel Corporation. */
 
 #pragma once
+#include <memkind_memtier.h>
```
Comment (michalbiesek) [inline tiering/ctl.h:5]: Not needed I believe (I just look into the last commit)

File: tiering/ctl.h [hunk 2]
```diff
@@ -2,10 +2,21 @@
...
+
+#include <stddef.h>
+
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+typedef struct tier_draft {
```
Comment (michalbiesek) [inline tiering/ctl.h:13]: Move this into ctl.c (I just look into the last commit)

File: tiering/ctl.c [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // SPDX-License-Identifier: BSD-2-Clause
 /* Copyright (C) 2021 Intel Corporation. */
 
-#include <memkind_memtier.h>
+#include <tiering/ctl.h>
```
Comment (michalbiesek) [inline tiering/ctl.c:4]: #include <memkind_memtier.h> should be sufficient here

**Review by bratpiorka — COMMENTED** [review]
> Reviewed 1 of 3 files at r2, 2 of 3 files at r3, 2 of 2 files at r4, 2 of 2 files at r5.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/589)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @PatKamin)

File: tiering/ctl.c [hunk 4]
```diff
@@ -279,25 +275,21 @@ static const char *ctl_policy_to_str(memtier_policy_t policy)
 
 struct memtier_kind *ctl_create_tier_kind_from_env(char *env_var_string)
 {
-    char *kind_name = NULL;
-    char *pmem_path = NULL;
     struct memtier_kind *tier_kind;
-    size_t pmem_size = 0;
-    unsigned ratio_value = 0;
+    static struct tier_draft tier;
```
Comment (michalbiesek) [inline tiering/ctl.c:279]: struct tier_draft tier = {NULL, NULL, 0 ,0}; - no need to be static

File: tiering/ctl.c [hunk 5]
```diff
@@ -154,38 +154,37 @@ static int ctl_parse_ratio(const char *ratio_str, unsigned *dest)
...
+        tier->pmem_path = strtok_r(NULL, CTL_VALUE_SEPARATOR, &sptr);
+        if (tier->pmem_path == NULL) {
             return -1;
         }
         char *pmem_size_str = strtok_r(NULL, CTL_VALUE_SEPARATOR, &sptr);
         if (pmem_size_str == NULL) {
             return -1;
         }
-        ret = ctl_parse_pmem_size(pmem_size_str, pmem_size_value);
+        ret = ctl_parse_pmem_size(pmem_size_str, &(tier->pmem_size));
```
Comment (michalbiesek) [inline tiering/ctl.c:179]: &(tier->pmem_size) -> &tier->pmem_size

File: tiering/ctl.c [hunk 6]
```diff
@@ -154,38 +154,37 @@ static int ctl_parse_ratio(const char *ratio_str, unsigned *dest)
...
         if (ret != 0) {
             log_err("Failed to parse pmem size: %s", pmem_size_str);
             return -1;
         }
     }
 
     char *ratio_value_str = strtok_r(NULL, CTL_VALUE_SEPARATOR, &sptr);
-    ret = ctl_parse_ratio(ratio_value_str, ratio_value);
+    ret = ctl_parse_ratio(ratio_value_str, &(tier->ratio_value));
```
Comment (michalbiesek) [inline tiering/ctl.c:187]: &(tier->ratio_value) -> &tier->ratio_value

File: tiering/ctl.h [hunk 7]
```diff
@@ -2,10 +2,21 @@
...
+    char *kind_name;
+    char *pmem_path;
+    size_t pmem_size;
+    unsigned ratio_value;
+} tier_draft;
```
Comment (michalbiesek) [inline tiering/ctl.h:18]: tier_draft -> ctl_tier_cfg ? not sure something with control and tier configuration ?

**Review by PatKamin — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/589)* status: all files reviewed, 7 unresolved discussions (waiting on @michalbiesek)

___
*[tiering/ctl.h, line 5 at r5](https://reviewable.io/reviews/memkind/memkind/589#-MWTdzVwT2hWKAMYXiu4-r5-5:-MWUAy_JEPg20EyD6U1T:b-d4gjs6) ([raw file](https://github.com/memkind/memkind/blob/ed0633d4bed89146a9ef1dc91c5d63e4629840e3/tiering/ctl.h#L5)):*
<details><summary><i>Previously, michalbiesek wrote…</i></summary><blockquote>

Not needed I believe (I just look into the last commit)
</blockquote></details>

Moved to .c file.
___
*[tiering/ctl.h, line 13 at r5](https://reviewable.io/reviews/memkind/memkind/589#-MWTdzVwT2hWKAMYXiu4-r5-13:-MWUBDi1-ORkbEmJLZ2l:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/ed0633d4bed89146a9ef1dc91c5d63e4629840e3/tiering/ctl.h#L13)):*
<details><summary><i>Previously, michalbiesek wrote…</i></summary><blockquote>

Move this into ctl.c (I just look into the last commit)
</blockquote></details>

Done.
___
*[tiering/ctl.h, line 18 at r5](https://reviewable.io/reviews/memkind/memkind/589#-MWTdzVwT2hWKAMYXiu4-r5-18:-MWUBTqH8wLyDBT9_LNC:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/ed0633d4bed89146a9ef1dc91c5d63e4629840e3/tiering/ctl.h#L18)):*
<details><summary><i>Previously, michalbiesek wrote…</i></summary><blockquote>

tier_draft -> ctl_tier_cfg ? not sure something with control and tier configuration ?
</blockquote></details>

Done.
___
*[tiering/ctl.c, line 4 at r5](https://reviewable.io/reviews/memkind/memkind/589#-MWTdzVwT2hWKAMYXiu3-r5-4:-MWUBUw1-WDSQpfocZJr:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/ed0633d4bed89146a9ef1dc91c5d63e4629840e3/tiering/ctl.c#L4)):*
<details><summary><i>Previously, michalbiesek wrote…</i></summary><blockquote>

#include <memkind_memtier.h> should be sufficient here
</blockquote></details>

Done.
___
*[tiering/ctl.c, line 179 at r5](https://reviewable.io/reviews/memkind/memkind/589#-MWTdzVwT2hWKAMYXiu3-r5-179:-MWUBfgy4FS16dGTYmJO:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/ed0633d4bed89146a9ef1dc91c5d63e4629840e3/tiering/ctl.c#L179)):*
<details><summary><i>Previously, michalbiesek wrote…</i></summary><blockquote>

&(tier->pmem_size) -> &tier->pmem_size
</blockquote></details>

Done.
___
*[tiering/ctl.c, line 187 at r5](https://reviewable.io/reviews/memkind/memkind/589#-MWTdzVwT2hWKAMYXiu3-r5-187:-MWUBgBR636pnkBPPzpk:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/ed0633d4bed89146a9ef1dc91c5d63e4629840e3/tiering/ctl.c#L187)):*
<details><summary><i>Previously, michalbiesek wrote…</i></summary><blockquote>

&(tier->ratio_value) -> &tier->ratio_value
</blockquote></details>

Done.
___
*[tiering/ctl.c, line 279 at r5](https://reviewable.io/reviews/memkind/memkind/589#-MWTdzVwT2hWKAMYXiu3-r5-279:-MWUC8x26VtdERvuHct5:b-896fix) ([raw file](https://github.com/memkind/memkind/blob/ed0633d4bed89146a9ef1dc91c5d63e4629840e3/tiering/ctl.c#L279)):*
<details><summary><i>Previously, michalbiesek wrote…</i></summary><blockquote>

struct tier_draft tier = {NULL, NULL, 0 ,0}; - no need to be static
</blockquote></details>

Done.

**Review by michalbiesek — APPROVED** [review]
> Reviewed 2 of 2 files at r6.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/589)* status: all files reviewed, 2 unresolved discussions (waiting on @PatKamin)

___
*[tiering/ctl.c, line 15 at r6](https://reviewable.io/reviews/memkind/memkind/589#-MWUEo5cDqaDwkiy_PVX:-MWUEo5cDqaDwkiy_PVY:bm56v6g) ([raw file](https://github.com/memkind/memkind/blob/5f0a14c8735508f0571c3f087f94c49638d69a0f/tiering/ctl.c#L15)):*
> ```C
> #include <string.h>
> 
> // TODO - remove this after logging cleanup
> ```

nit: please change the pull request title to be inline with last commit messgae
___
*[tiering/ctl.c, line 21 at r6](https://reviewable.io/reviews/memkind/memkind/589#-MWUEDWN3ibvaCwwuqkE:-MWUEDWN3ibvaCwwuqkF:brzuxnv) ([raw file](https://github.com/memkind/memkind/blob/5f0a14c8735508f0571c3f087f94c49638d69a0f/tiering/ctl.c#L21)):*
> ```
> static struct memtier_tier *current_tier;
> typedef struct ctl_tier_cfg {
> ```
nit: split this with a newline

**Review by bratpiorka — COMMENTED** [review]
:lgtm:

> Reviewed 1 of 1 files at r7.
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/589#-:-MWUPt9dC9HR2ceqDH_g:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @PatKamin)

**Comment by michalbiesek** [comment]
Rebase please

**Review by michalbiesek — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/memkind/memkind/589)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @PatKamin)

**Pull request merged**